get_tracking_numbers.py

```diff
@@ -31,71 +45,74 @@ def main():
   email_sender = EmailSender(email_config)
 
   print("Retrieving Amazon tracking numbers from email...")
-  amazon_tracking_retriever = AmazonTrackingRetriever(config, driver_creator)
+  amazon_tracking_retriever = AmazonTrackingRetriever(config, args, driver_creator)
   try:
     trackings = amazon_tracking_retriever.get_trackings()
   except:
     send_error_email(email_sender, "Error retrieving Amazon emails")
     raise
 
+  action_taken = "" if args.seen else " and marked them as unread"
   if amazon_tracking_retriever.failed_email_ids:
-    print(
-        "Found %d Amazon emails without buying group labels and marked them as unread. Continuing..."
-        % len(amazon_tracking_retriever.failed_email_ids))
+    print(f"Found {len(amazon_tracking_retriever.failed_email_ids)} Amazon emails "
+          f"without buying group labels{action_taken}. Continuing...")
 
   print("Retrieving Best Buy tracking numbers from email...")
-  bestbuy_tracking_retriever = BestBuyTrackingRetriever(config, driver_creator)
+  bestbuy_tracking_retriever = BestBuyTrackingRetriever(config, args, driver_creator)
   try:
-    bb_trackings = bestbuy_tracking_retriever.get_trackings()
-    trackings.extend(bb_trackings)
+    trackings.update(bestbuy_tracking_retriever.get_trackings())
   except:
     send_error_email(email_sender, "Error retrieving BB emails")
     raise
 
   if bestbuy_tracking_retriever.failed_email_ids:
-    print(
-        "Found %d BB emails without buying group labels and marked them as unread. Continuing..."
-        % len(bestbuy_tracking_retriever.failed_email_ids))
+    print(f"Found {len(bestbuy_tracking_retriever.failed_email_ids)} Best Buy emails "
+          f"without buying group labels{action_taken}. Continuing...")
 
   try:
-    print("Found %d total tracking numbers" % len(trackings))
+    tracking_output = TrackingOutput(config)
+    existing_tracking_nos = set([t.tracking_number for t in tracking_output.get_existing_trackings()])
+    new_tracking_nos = set(trackings.keys()).difference(existing_tracking_nos)
```

Comment:
This means that we won't include old tracking numbers in the email notifications if you mark emails as unread and re-run. This is definitely a difference in behavior, though I'm not sure it's necessarily bad. My main concern is BFMR -- if we forget to upload something 

Of course, BFMR trackings still appear in the console and on the sheet

Reply:
There's a bigger issue here, which is that just because we're aware of a tracking # doesn't mean that the BG is.  Maybe uploading glitched out at some point in time for some reason, and the BG didn't actually save it and we couldn't detect that failure through WebDriver.  So the safe thing to do would be to continue to try to upload every tracking # that we're aware of.

But in the emails and what's displayed on screen, it definitely makes sense to only display what's new.  Otherwise it's way too noisy, especially when searching your inbox for a given tracking or order #; I don't want to see every run it was picked up on, only the first.

As for BFMR specifically, the real solution is to automate uploading of tracking #s.  Relying on manually copy-pasting them from the notification email is way too error-prone.  This one's on you.